prov/sockets/src/sock_progress.c

```diff
@@ -76,6 +76,10 @@ static inline ssize_t sock_pe_send_field(struct sock_pe_entry *pe_entry,
 
 	offset = pe_entry->done_len - start_offset;
 	data_len = field_len - offset;
+	if (!field && pe_entry->pe.rx.rx_op.atomic.op == FI_ATOMIC_READ) {
+		pe_entry->done_len += field_len;
+		return 0;
+	}
```

Comment:
Can we move this check inside tx_atomic and skip the call to sock_pe_send_field?

sock_pe_send_field() is common for all the operations and it might not be good to have this check here.